src/main/java/com/simibubi/create/content/kinetics/mechanicalArm/ArmBlockEntity.java

```diff
@@ -142,12 +148,24 @@ public void tick() {
 		if (level.isClientSide)
 			return;
 
-		if (phase == Phase.MOVE_TO_INPUT)
-			collectItem();
-		else if (phase == Phase.MOVE_TO_OUTPUT)
-			depositItem();
-		else if (phase == Phase.SEARCH_INPUTS || phase == Phase.DANCING)
-			searchForItem();
+		switch (phase) {
+			case MOVE_TO_INPUT:
+				collectItem();
+				break;
+			case MOVE_TO_OUTPUT:
+				depositItem();
+				break;
+			case SEARCH_INPUTS:
+			case DANCING:
+				boolean forcedSearch = forceSearchInputsTicks > 0;
+				if (forcedSearch || (level.getGameTime() + randomSearchTickOffset) % SearchInputsRate == 0) {
```

Comment:
Should probably make it clear in the comments above that the offset will be modulo'd and explain why